internal/services/datafactory/data_factory_dataset_binary_resource.go

```diff
@@ -229,16 +230,21 @@ func resourceDataFactoryDatasetBinary() *pluginsdk.Resource {
 						},
 						// SFTP Specific field
 						"type": {
-							Type:     pluginsdk.TypeString,
-							Required: true,
+							Type:             pluginsdk.TypeString,
+							Required:         true,
+							DiffSuppressFunc: suppress.CaseDifference,
 							ValidateFunc: validation.StringInSlice([]string{
-								string(datafactory.TypeBasicDatasetCompressionTypeBZip2),
-								string(datafactory.TypeBasicDatasetCompressionTypeDeflate),
-								string(datafactory.TypeBasicDatasetCompressionTypeGZip),
-								string(datafactory.TypeBasicDatasetCompressionTypeTar),
-								string(datafactory.TypeBasicDatasetCompressionTypeTarGZip),
-								string(datafactory.TypeBasicDatasetCompressionTypeZipDeflate),
-							}, false),
+								string(datafactory.CompressionCodecBzip2),
+								string(datafactory.CompressionCodecDeflate),
+								string(datafactory.CompressionCodecGzip),
+								string(datafactory.CompressionCodecLz4),
+								string(datafactory.CompressionCodecLzo),
+								string(datafactory.CompressionCodecNone),
```

Comment:
this generally isn't a valid value.. can we either prove this value works or remove this

```suggestion
```

Reply:
Thanks, I will remove it